packages/next-swc/crates/next-core/src/next_shared/transforms/next_page_config.rs

```diff
@@ -32,15 +35,14 @@ pub fn get_next_page_config_rule(enable_mdx_rs: bool, is_page_file: bool) -> Mod
 #[derive(Debug)]
 struct NextPageConfig {
     is_development: bool,
-    is_page_file: bool,
 }
 
 #[async_trait]
 impl CustomTransformer for NextPageConfig {
     async fn transform(&self, program: &mut Program, _ctx: &TransformContext<'_>) -> Result<()> {
         let p = std::mem::replace(program, Program::Module(Module::dummy()));
 
-        *program = p.fold_with(&mut page_config(self.is_development, self.is_page_file));
```

Comment:
I expected is_page_file would do the guarding for the pages.

Reply:
It does, but you passed `pages_dir.is_some()` which only check if there is a pages_dir and not if the file is a page file